test/button.test.tsx

```diff
@@ -1,11 +1,265 @@
-import { test, Page} from "bigtest";
-import { Button as btButton } from "../src/index";
-import { Button as MyButton } from "@material-ui/core";
+import { test, Page, Link } from "bigtest";
+import { Button as Interactor, including, not, or } from "../src/index";
+import { Button as MuiButton } from "@material-ui/core";
+import { containing } from './matchers';
 import { render } from "./helpers";
 
+const button = Interactor();
+
 export default test("Button")
- .step(Page.visit("/"))
- .step("rendering a button", async() => {
-   render(<MyButton>Click me</MyButton>)
- })
- .assertion(btButton("CLICK ME").exists())
\ No newline at end of file
+  .step(Page.visit("/"))
+  .child("rendering a button", (test) => test
+    .step("render button", async() => {
+      await render(<MuiButton>My Button</MuiButton>)
+    })
+    .assertion(button.exists())
+    .assertion(button.has({ text: "My Button".toUpperCase()}))
+
+    .assertion("should render with the root, text and textPrimary classes but no others", async() => {
+    await button.has({ classList: containing(["MuiButton-root", "MuiButton-text"]) })
+    await button.has({ classList: not(containing(["MuiButton-textSecondary", "MuiButton-outlined", "MuiButton-contained"])) })
+    })
+  )
+  .child('render a text secondary button', (test) => test
+    .step("render secondary button", async() =>{
+      await render(<MuiButton color="secondary">Secondary Button</MuiButton>)
+    })
+    .assertion(button.has({ className: including("MuiButton-root")}))
```

Comment:
@wKich I reverted the file back to just use classNames and removed the dependency on unstyled. This is how I initially started so I was in my history.